### ISSUE TITLE
Chopper trap detection/ precision

### ISSUE THREAD

**Issue by ace-dent** [Press-Play-On-Tape/PrinceOfArabia#139]
Tested with v1.01 Beta 008, Sim.

- L7. Running passed chopper trap, seems safe.
-  Then moving off again (starting to run), trap kills the prince.

![recording_20230327012420](https://user-images.githubusercontent.com/473224/227811599-5c3d5050-e184-421a-b0b7-c2a62de1c858.gif)

**Comment by filmote** [comment]
Yeah.  The choppers have been really problematic.  I hate them.  But will definitely look at this.

**Comment by ace-dent** [comment]
Tested with Version 1.01 Beta 012, FX Hardware.
- This seems better and in general 'fair'.
- There is still the edge case, where your position can be _inside_ the trap... as long as you don't move at the wrong time! (see recording - here I'm in the danger zone and squatting at different times).
- If changing/ adding logic adds too much code, current solution is fine and 'Fixed'.

![Chop1](https://user-images.githubusercontent.com/473224/229195306-cc3ab214-0163-48fa-a895-058310a0ebe3.gif)

**Comment by ace-dent** [comment]
Tested with Version 1.01 Beta 012, FX Hardware.
- L9: See recording.
- Trying to jump through at the wrong time, leads to prince dying on the spot (where `A` was pressed).
- Would expect corpse to appear at the blades, not some distance away.

![Distanttrap](https://user-images.githubusercontent.com/473224/229245829-3562ac93-6268-48a0-9aa0-73abfb330ffa.gif)

**Comment by ace-dent** [comment]
Tested with Version 1.01 Beta 012, Sim
- L9 (and earlier).
- Prince's sword should be ignored for trap detection. 
- Only consider princes' body for collsion.

![Blades](https://user-images.githubusercontent.com/473224/229250079-2833708a-8f5c-4e57-89ce-b8c1368c3470.gif)

**Comment by filmote** [comment]
'Trying to jump through at the wrong time, leads to prince dying on the spot (where A was pressed).'

Well .. its not as simple as that.  I have a table that describes for each image the reach of the player and the front and back foot position.  These are used for all sorts of things like determining if you are standing on a platform, whether you are about to fall or whether you can move forward or not.

When you are running, the head, body and arms move relative to the body's centerline.  So when you run through the blades a part of you must hit the blade but when you die, you fall down relative to the centreline of the body.  Hence that is why sometimes you fall dead away from teh blade.  It all depends on which image you touch the blade.

I have changed the table to remove the reach of the sword itself.  Hope this helps the issue.
